lib/models/cosmos_wallet.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:sacco/wallet.dart';
+
+class CosmosWallet extends BaseWalletDetails {
+  Wallet wallet;
+
+  CosmosWallet({
+    required String walletAddress,
+    required String walletAlias,
+    WalletType walletType = WalletType.Cosmos,
+    required this.wallet,
+  }) : super(
+          walletAlias: walletAlias,
+          walletAddress: walletAddress,
+          walletType: walletType,
+        );
+}
+
+class BaseWalletDetails {
+  String walletAddress;
```

Comment:
all the classes that are merely holding data should be immutable as much as possible, thus all the fields should be marked as final. we can then also mark the constructor with `const` keyword.

We can also rename the class to `WalletDetails`, no need for the `Base` prefix

Reply:
Prefix is used because it is the parent class of different types of wallets. Cosmos and Ethereum wallets extend this class.